nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/bigquery/PutBigQueryBatch.java

```diff
@@ -0,0 +1,269 @@
+
+        final BigQuery bq = getCloudService();
+
+        final String projectId = context.getProperty(PROJECT_ID).evaluateAttributeExpressions().getValue();
+        final String dataset = context.getProperty(DATASET).evaluateAttributeExpressions(flowFile).getValue();
+        final String tableName = context.getProperty(TABLE_NAME).evaluateAttributeExpressions(flowFile).getValue();
+
+        final TableId tableId;
+        if (StringUtils.isEmpty(projectId)) {
+            tableId = TableId.of(dataset, tableName);
+        } else {
+            tableId = TableId.of(projectId, dataset, tableName);
+        }
+
+        final String fileType = context.getProperty(SOURCE_TYPE).getValue();
+
+        String schemaString = context.getProperty(TABLE_SCHEMA).evaluateAttributeExpressions().getValue();
+        Schema schema = BqUtils.schemaFromString(schemaString);
+
+        WriteChannelConfiguration writeChannelConfiguration =
+                WriteChannelConfiguration.newBuilder(tableId)
+                        .setCreateDisposition(JobInfo.CreateDisposition.valueOf(context.getProperty(CREATE_DISPOSITION).getValue()))
+                        .setWriteDisposition(JobInfo.WriteDisposition.valueOf(context.getProperty(WRITE_DISPOSITION).getValue()))
+                        .setIgnoreUnknownValues(context.getProperty(IGNORE_UNKNOWN).asBoolean())
+                        .setMaxBadRecords(context.getProperty(MAXBAD_RECORDS).asInteger())
+                        .setSchema(schema)
+                        .setFormatOptions(FormatOptions.of(fileType))
+                        .build();
+
+        TableDataWriteChannel writer = bq.writer(writeChannelConfiguration);
+
+        try {
+            session.read(flowFile, rawIn -> {
+                ReadableByteChannel readableByteChannel = Channels.newChannel(rawIn);
+                ByteBuffer byteBuffer = ByteBuffer.allocateDirect(BUFFER_SIZE);
+                while (readableByteChannel.read(byteBuffer) >= 0) {
+                    byteBuffer.flip();
+                    writer.write(byteBuffer);
+                    byteBuffer.clear();
+                }
+            });
+
+            writer.close();
+
+            Job job = writer.getJob();
+            PropertyValue property = context.getProperty(READ_TIMEOUT);
+            Long timePeriod = property.evaluateAttributeExpressions().asTimePeriod(TimeUnit.SECONDS);
+            Duration duration = Duration.of(timePeriod, ChronoUnit.SECONDS);
+            job = job.waitFor(RetryOption.totalTimeout(duration));
+
+            if (job != null) {
+                attributes.put(BigQueryAttributes.JOB_CREATE_TIME_ATTR, Long.toString(job.getStatistics().getCreationTime()));
+                attributes.put(BigQueryAttributes.JOB_END_TIME_ATTR, Long.toString(job.getStatistics().getEndTime()));
+                attributes.put(BigQueryAttributes.JOB_START_TIME_ATTR, Long.toString(job.getStatistics().getStartTime()));
```

Comment:
In case of job success, could we add the number of records we added in the table, using:
````java
LoadStatistics stats = (LoadStatistics) job.getStatistics();
stats.getOutputRows();
````